File: submissions/jbk/week8/version1-django/notes/management/commands/seed.py
from django.core.management.base import BaseCommand
from notes.models import ActionItem, Note


class Command(BaseCommand):
    help = "Seed the database with sample data"

    def handle(self, *args, **options):
        Note.objects.all().delete()
        ActionItem.objects.all().delete()

        Note.objects.create(
            title="Welcome",
            content="This is a starter note. TODO: explore the app!",
        )
        Note.objects.create(
            title="Demo",
            content="Click around and add a note. Ship feature!",
        )
        ActionItem.objects.create(description="Try pre-commit", completed=False)
        ActionItem.objects.create(description="Run tests", completed=False)

        self.stdout.write(self.style.SUCCESS("Seeded 2 notes and 2 action items"))

File: submissions/jbk/week8/version2-mern/server/seed.js
require("dotenv").config();
const mongoose = require("mongoose");
const Note = require("./models/Note");
const ActionItem = require("./models/ActionItem");

const MONGODB_URI =
  process.env.MONGODB_URI || "mongodb://localhost:27017/notesapp";

async function seed() {
  await mongoose.connect(MONGODB_URI);
  console.log("Connected to MongoDB");

  await Note.deleteMany({});
  await ActionItem.deleteMany({});

  await Note.create([
    {
      title: "Welcome",
      content: "This is a starter note. TODO: explore the app!",
    },
    { title: "Demo", content: "Click around and add a note. Ship feature!" },
  ]);

  await ActionItem.create([
    { description: "Try pre-commit", completed: false },
    { description: "Run tests", completed: false },
  ]);

  console.log("Seeded 2 notes and 2 action items");
  await mongoose.disconnect();
}

seed().catch(console.error);
